multicluster/service-mirror/cluster_watcher.go

```diff
@@ -324,19 +356,49 @@ func (rcsw *RemoteClusterServiceWatcher) cleanupMirroredResources(ctx context.Co
 // Deletes a locally mirrored service as it is not present on the remote cluster anymore
 func (rcsw *RemoteClusterServiceWatcher) handleRemoteServiceDeleted(ctx context.Context, ev *RemoteServiceDeleted) error {
 	localServiceName := rcsw.mirroredResourceName(ev.Name)
-	rcsw.log.Infof("Deleting mirrored service %s/%s", ev.Namespace, localServiceName)
+	localService, err := rcsw.localAPIClient.Svc().Lister().Services(ev.Namespace).Get(localServiceName)
 	var errors []error
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			rcsw.log.Debugf("Failed to delete mirror service %s/%s: %v", ev.Namespace, ev.Name, err)
+			return nil
+		}
+		errors = append(errors, fmt.Errorf("could not fetch service %s/%s: %s", ev.Namespace, localServiceName, err))
+	}
+
+	// If the mirror service is headless, also delete its endpoint mirror
+	// services.
+	if rcsw.headlessServicesEnabled && localService.Spec.ClusterIP == corev1.ClusterIPNone {
+		matchLabels := map[string]string{
+			consts.MirroredHeadlessSvcNameLabel: localServiceName,
+		}
+		endpointMirrorServices, err := rcsw.localAPIClient.Svc().Lister().List(labels.Set(matchLabels).AsSelector())
+		if err != nil {
+			errors = append(errors, fmt.Errorf("could not fetch endpoint mirrors for service %s/%s: %s", ev.Namespace, localServiceName, err))
```

Comment:
I think it would make sense to only include the errors if the service is not found so going to add that in. Also makes sense to make the distinction in the log output!